xbmc/cores/VideoPlayer/VideoRenderers/OverlayRenderer.cpp

```diff
@@ -432,7 +432,11 @@ COverlay* CRenderer::Convert(CDVDOverlay* o, double pts)
 #endif
 
   if(!r && o->IsOverlayType(DVDOVERLAY_TYPE_TEXT))
-    r = new COverlayText(static_cast<CDVDOverlayText*>(o));
+  {
+    int targetWidth = MathUtils::round_int(m_rv.Width());
+    int targetHeight = MathUtils::round_int(m_rv.Height());
+    r = new COverlayText(static_cast<CDVDOverlayText*>(o), targetWidth, targetHeight);
```

Comment:
passing changeable parameters with ctor is not a good idea. overlay is only converted once and view window may change.

Reply:
I completely forgot about this (and did it already before with the `bgcolor`). Last question before I start refactoring. I understand the parameters must be passed to `PrepareRender`. However the method has already 7 arguments:

```
void COverlayText::PrepareRender(const std::string &font, int color, int height, int style, const std::string &fontcache, const std::string &fontbordercache, const UTILS::Color bgcolor)
```

Is there any way to make this more elegant in C++? Maybe create a struct and pass it to the method instead of all those arguments?

Reply:
I don't have a good idea for this. IMO no big deal if you pass that many arguments